roseau/load_flow/models/loads/loads.py

```diff
@@ -223,15 +231,18 @@ def powers(self) -> np.ndarray:
     @ureg.wraps(None, (None, "VA"), strict=False)
     def powers(self, value: Sequence[complex]) -> None:
         self._powers = self._validate_value(value)
+        self._invalidate_network()
 
     @property
     def res_flexible_powers(self) -> np.ndarray:
         """The load flow result of the load flexible powers (VA)."""
-        if self._res_flexible_powers is None:
-            self._raise_load_flow_not_run()
-        return self._res_flexible_powers
+        return self._res_getter(value=self._res_flexible_powers, warning=True)
 
     def to_dict(self) -> JsonDict:
+        if self.bus is None:
+            msg = f"The element {self!r} is disconnected and can not be used anymore."
```

Comment:
```suggestion
            msg = f"The load {self.id!r} is disconnected and can not be used anymore."
```
I think this is a better message for the user but no strong opinion.